curator-framework/src/main/java/org/apache/curator/framework/imps/EnsembleTracker.java

```diff
@@ -166,12 +166,26 @@ public static String configToConnectionString(QuorumVerifier data) throws Except
         StringBuilder sb = new StringBuilder();
         for ( QuorumPeer.QuorumServer server : data.getAllMembers().values() )
         {
+            if ( server.clientAddr == null )
+            {
+                // Invalid client address configuration in zoo.cfg
+                continue;
```

Comment:
The spec of server address is server.<positive id> = <address1>:<port1>:<port2>[:role];[<client port address>:]<client port>, and if server.clientAddr is null then it means the client port address is not present. The client port address is optional, so the address is still a valid address.